pkgs/by-name/at/atf/package.nix

```diff
@@ -47,6 +50,14 @@ stdenv.mkDerivation (finalAttrs: {
     "CXXFLAGS=-std=c++14"
   ];
 
+  # Needed for cross compilation to avoid check errors
+  # https://github.com/NixOS/nixpkgs/issues/413910
+  configureFlags = lib.optionals (!stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
+    "kyua_cv_getopt_plus=yes"
+    "kyua_cv_attribute_noreturn=yes"
+    "kyua_cv_getcwd_works=yes"
```

Comment:
to get this using the host `bash`, add to configureFlags:
```
  "ATF_SHELL=${lib.getExe bashNonInteractive}"
```

I'm not sure the solution on darwin: if they have `/bin/sh`, you could just use that instead of `bashNonInteractive`?

Reply:
`/bin/sh` is impure and also a really old version of Bash.

This probably needs a change to the stdenv bootstrap to use Bash from a previous stage to break the infinite recursion. Alternatively, try only adding `bash` to `buildInputs` when cross-compiling.

Reply:
> This probably needs a change to the stdenv bootstrap to use Bash from a previous stage to break the infinite recursion.

I don't know how to do this, how to proceed?

> Alternatively, try only adding bash to buildInputs when cross-compiling.

Is `stdenv.buildPlatform != stdenv.targetPlatform` the way to go?